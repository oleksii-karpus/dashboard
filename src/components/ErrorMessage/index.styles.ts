import { Box, styled } from '@mui/material';

export const ErrorListStyled = styled(Box)`
    display: flex;
    flex-direction: column;
    margin: 10px 0;
`;

export const ErrorItemStyled = styled(Box)`
    color: ${({ theme }) => theme.palette.error.light};
`;
