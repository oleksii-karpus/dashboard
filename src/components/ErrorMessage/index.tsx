import { ErrorItemStyled, ErrorListStyled } from './index.styles';

type Props = {
    errors?: string[];
};
export const ErrorMessage = ({ errors }: Props) => {
    if (!errors || !errors.length) {
        return null;
    }
    return (
        <ErrorListStyled>
            {errors.map(error => (
                <ErrorItemStyled key={error}>{error}</ErrorItemStyled>
            ))}
        </ErrorListStyled>
    );
};
