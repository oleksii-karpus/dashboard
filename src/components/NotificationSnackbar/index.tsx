import { Snackbar, Alert } from '@mui/material';
import { useSelector, useDispatch } from 'react-redux';
import { RootState } from '../../store/types/root.state';
import { notificationsRoutine } from '../../store/app/routines';
import { getAppState } from '../../utils/get.state.utils';

const NotificationSnackbar = () => {
    const { message, severity, open } = useSelector((state: RootState) => getAppState(state).notification);
    const dispatch = useDispatch();

    const handleClose = () => {
        dispatch(notificationsRoutine.fulfill());
    };

    return (
        <Snackbar
            open={open}
            autoHideDuration={6000}
            onClose={handleClose}
            anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        >
            <Alert severity={severity} onClose={handleClose}>
                {message}
            </Alert>
        </Snackbar>
    );
};

export default NotificationSnackbar;
