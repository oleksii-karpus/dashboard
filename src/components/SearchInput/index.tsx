import SearchIcon from '@mui/icons-material/Search';
import { ChangeEvent } from 'react';
import { InputBaseStyled, SearchIconWrapperStyled, WrapperStyled } from './index.styles';

type Props = {
    onClick: () => void;
    searchQuery: string;
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
};

export const SearchInput = ({ onClick, searchQuery, onChange }: Props) => {
    return (
        <WrapperStyled>
            <SearchIconWrapperStyled onClick={onClick}>
                <SearchIcon />
            </SearchIconWrapperStyled>
            <InputBaseStyled
                value={searchQuery}
                onChange={onChange}
                placeholder="Searchâ€¦"
                inputProps={{ 'aria-label': 'search' }}
            />
        </WrapperStyled>
    );
};
