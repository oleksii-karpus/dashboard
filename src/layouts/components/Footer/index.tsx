import { WrapperStyled } from './index.styles';

export const Footer = () => <WrapperStyled>Footer</WrapperStyled>;
