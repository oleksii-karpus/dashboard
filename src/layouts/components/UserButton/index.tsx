import React from 'react';
import { IconButton, Menu, MenuItem } from '@mui/material';
import { AccountCircle } from '@mui/icons-material';

type Props = {
    onLogout: () => void;
};

export const UserButton = ({ onLogout }: Props) => {
    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

    const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
        setAnchorEl(event.currentTarget);
    };

    const handleClose = () => {
        setAnchorEl(null);
    };

    const handleLogout = () => {
        onLogout();
        setAnchorEl(null);
    };

    return (
        <>
            <IconButton
                size="large"
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={handleMenu}
                color="inherit"
            >
                <AccountCircle />
            </IconButton>
            <Menu
                id="menu-appbar"
                anchorEl={anchorEl}
                anchorOrigin={{
                    vertical: 'top',
                    horizontal: 'right'
                }}
                keepMounted
                transformOrigin={{
                    vertical: 'top',
                    horizontal: 'right'
                }}
                open={Boolean(anchorEl)}
                onClose={handleClose}
            >
                <MenuItem onClick={handleLogout}>Logout</MenuItem>
            </Menu>
        </>
    );
};
