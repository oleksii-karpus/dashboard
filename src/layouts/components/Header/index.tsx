import { AppBar, Box, Toolbar } from '@mui/material';
import { useLocation } from 'react-router-dom';
import { Routes } from '../../../common/enums/routes';
import { NavItem } from '../../types/nav.items';
import { useAuth } from '../../../hooks/use.auth';
import { UserButton } from '../UserButton';
import NotificationSnackbar from '../../../components/NotificationSnackbar';
import { LogoWrapper, NavItemLinkStyled, NavStyled } from './index.styles';

type Props = {
    navItems: NavItem[];
};

export const Header = ({ navItems }: Props) => {
    const { user, onLogout } = useAuth();
    const location = useLocation();
    const isActive = (route: string): boolean => {
        const pathSegments = location.pathname.split('/').filter(Boolean);
        return pathSegments.includes(route);
    };
    return (
        <Box>
            <NotificationSnackbar />
            <AppBar position="static">
                <Toolbar>
                    <LogoWrapper>
                        <NavItemLinkStyled to={Routes.home}>Home</NavItemLinkStyled>
                    </LogoWrapper>
                    <NavStyled>
                        {navItems.map(item => (
                            <NavItemLinkStyled key={item.route} to={item.route} isActive={isActive(item.route)}>
                                {item.label}
                            </NavItemLinkStyled>
                        ))}
                        {user ? (
                            <UserButton onLogout={onLogout} />
                        ) : (
                            <NavItemLinkStyled key="login" to={Routes.login} isActive={isActive(Routes.login)}>
                                Login
                            </NavItemLinkStyled>
                        )}
                    </NavStyled>
                </Toolbar>
            </AppBar>
        </Box>
    );
};
