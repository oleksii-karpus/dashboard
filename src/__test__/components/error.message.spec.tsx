import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ErrorMessage } from '../../components/ErrorMessage';

describe('ErrorMessage', () => {
    it('matches snapshot when errors are provided', () => {
        const errors = ['Error 1', 'Error 2'];
        const { asFragment } = render(<ErrorMessage errors={errors} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot when no errors are provided', () => {
        const { asFragment } = render(<ErrorMessage errors={undefined} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot when an empty array is provided', () => {
        const { asFragment } = render(<ErrorMessage errors={[]} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders all error messages', () => {
        const errors = ['Error 1', 'Error 2'];
        render(<ErrorMessage errors={errors} />);

        errors.forEach(error => {
            expect(screen.getByText(error)).toBeInTheDocument();
        });
    });

    it('does not render error list when no errors are provided', () => {
        const { container } = render(<ErrorMessage errors={undefined} />);
        expect(container.firstChild).toBeNull();
    });

    it('does not render error list when an empty array is provided', () => {
        const { container } = render(<ErrorMessage errors={[]} />);
        expect(container.firstChild).toBeNull();
    });
});
