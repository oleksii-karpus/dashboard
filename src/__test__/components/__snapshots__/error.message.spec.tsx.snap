// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot when an empty array is provided 1`] = `<DocumentFragment />`;

exports[`ErrorMessage matches snapshot when errors are provided 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-vo4hhh"
  >
    <div
      class="MuiBox-root css-4zc23b"
    >
      Error 1
    </div>
    <div
      class="MuiBox-root css-4zc23b"
    >
      Error 2
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorMessage matches snapshot when no errors are provided 1`] = `<DocumentFragment />`;
